### PR TITLE
suppress errors related to legacy client in code generator

### PR DESCRIPTION
**Pull request by shahrzadav**
Signed-off-by: Shahrzad <shahrzad.asghari73@gmail.com>

# Description

When the user selects Legacy client in code generator in secure communication task, errors related to this constraint will be suppressed, if and only if the checkbox related to this in preference page is also selected.

Fixes # (193)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

It has been tested in runtime.

**Test Configuration**:
* Eclipse Version: Eclipse Version: 2019-09 R (4.13.0)
* Java Version: 1.8
* OS: Windows 10

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/wizard/Configuration.java [hunk 1]
```diff
@@ -32,11 +38,27 @@
 
 	final protected Map<Question, Answer> options;
 	final protected String pathOnDisk;
+	protected Answer answr;
+	final protected String taskName;
 
 	@SuppressWarnings("unchecked")
-	public Configuration(Map<?, ?> constraints, String pathOnDisk) {
+	public Configuration(Map<?, ?> constraints, String pathOnDisk, String taskName) throws IOException {
```
Comment (AnakinRaW) [inline plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/wizard/Configuration.java]: what would happen if an IOException was thrown? Will the feature not work at all or would it be able to work regardless?

If it still would work, please catch the exception, log it and then forget it.

File: plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/results/ResultsCCUIListener.java [hunk 2]
```diff
@@ -118,9 +126,16 @@ public void analyzeDependenciesOnly(final Boolean depOnly) {
 	@Override
 	public void reportError(final AbstractError error) {
 		IResource sourceFile = null;
+		IPreferenceStore prefStore = Activator.getDefault().getPreferenceStore();
 		if (this.depOnly) {
 			return;
 		}
+		if (error instanceof ConstraintError) {
+			ConstraintError conError = (ConstraintError) error;
+			if (prefStore.getBoolean(Constants.SUPPRESS_LEGACYCLIENT_ERRORS) && conError.getBrokenConstraint().toString().equals("VC:protocol - TLSv1.2,") && isLegacyClient()) {
```
Comment (AnakinRaW) [inline plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/results/ResultsCCUIListener.java]: For extensibility reasons: please put the `"VC:protocol - TLSv1.2,"` string into some class that holds this as an constant.

File: plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/results/ResultsCCUIListener.java [hunk 3]
```diff
@@ -230,6 +245,27 @@ public void reportError(final AbstractError error) {
...
+	@SuppressWarnings("unchecked")
+	public Boolean isLegacyClient() {
+		final String path = this.currentProject.getLocation().toOSString() + Constants.innerFileSeparator + Constants.SECURE_COMMUNICATION + Constants.JSON_EXTENSION;
+		JSONParser parser = new JSONParser();
+		try {
+			if (Files.exists(Paths.get(path))) {
+
+					Object obj = parser.parse(new InputStreamReader(new FileInputStream(path)));
+					JSONObject jsonObject = (JSONObject) obj;
+					String clientsQuestion = "Which clients should the server at least support?";
```
Comment (AnakinRaW) [inline plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/results/ResultsCCUIListener.java]: please use string constants here that get used for this code and the JSON creation.

**Review by AnakinRaW — CHANGES_REQUESTED** [review]

**Comment by shahrzadav** [comment]
@schlichtig, could you please rebuild it?

**Comment by shahrzadav** [comment]
Made another PR and it is already merged. [https://github.com/eclipse-cognicrypt/CogniCrypt/pull/446](url)

**Pull request closed without merging**